ivy/functional/frontends/numpy/indexing_routines/indexing_like_operations.py

```diff
@@ -78,3 +79,24 @@ def diag(v, k=0):
 @to_ivy_arrays_and_back
 def diagonal(a, offset, axis1, axis2):
     return ivy.diagonal(a, offset=offset, axis1=axis1, axis2=axis2)
+
+
+@to_ivy_arrays_and_back
+@handle_numpy_out
+def compress(condition, a, *, axis=None, out=None):
```

Comment:
Hi! The signature needs to match the original function, so you'll have to remove `*`.
Also, we should always write a test for each function we are adding.